cmake/opentelemetry-proto.cmake

```diff
@@ -294,7 +294,7 @@ if(OPENTELEMETRY_INSTALL)
 endif()
 
 if(TARGET protobuf::libprotobuf)
-  target_link_libraries(opentelemetry_proto PUBLIC protobuf::libprotobuf)
```

Comment:
Why do we change the protobuf dependency to private here?The headers of opentelemetry_proto include the headers of protobuf and it also use some symbols in libprotobuf.

Reply:
I think this may have been the first change I made due to failing CMake configure. 
It looks like it might have been caused due to the below link of opentelemetry_proto being public, so if that is private maybe this is not needed.
I'll try it and see.
It's certainly true that if the header file include protobuf headers then you at least need an interface dependency.